posts/wordpress-to-jamstack.md

```diff
@@ -0,0 +1,152 @@
+
+In NextJS it gives you the ability to fetch data for static generation using `getStaticProps`. This is where you define the static content that you want your page to use and what the TinaCMS form will use.
+
+```javascript
+export const getStaticProps = async ({ preview, previewData }) => {
+  if (preview) {
+    return getGithubPreviewProps({
+      ...previewData,
+      fileRelativePath: "content/im-new.json",
+      parse: parseJson,
+    });
+  }
+  return {
+    props: {
+      sourceProvider: null,
+      error: null,
+      preview: false,
+      file: {
+        fileRelativePath: "content/im-new.json",
+        data: (await import("../content/im-new.json")).default,
+      },
+    },
+  };
+};
+```
+
+This then allows you to use `file.data` in your component along with the TinaCMS form React hook to link your static content with the TinaCMS editor which looks like the image below.
+
+![TinaCMS on website](https://raw.githubusercontent.com/blandfried/blog-usa/master/images/2021/04/tinacms.png)
+
+# Automating Sermon Videos
+
+On the Wordpress site they had to manually update the sermon video section every Monday and I decided that this could be an automated task so they would not have to worry about it. I ended up accomplishing this automation by using the serverless function capabilities that come with using NextJS and hosting it using Vercel. I wrote a Node JS function that calls the YouTube API to retrieve the five latest uploads. To save on api call quotas I made the decision to store the response data in a database since new sermons are uploaded on Sundays and do not change for a week. I figured there was no need to call the YouTube api anytime someone visited the site.
+
+I setup a cron job using [EasyCron](https://www.easycron.com/) that runs on Sunday mornings when the sermons begin and this calls the YouTube API that I created to fetch the latest videos. Since they have a live stream happening I display the live stream on the website so members do not have to go to YouTube to view the sermons and can watch on the website directly.
```

Comment:
```suggestion
I set up a cron job using [EasyCron](https://www.easycron.com/) that runs on Sunday mornings when the sermons begin, and this calls the YouTube API that I created to fetch the latest videos. Since they have a live stream happening, I display the live stream on the website so members do not have to go to YouTube to view the sermons and can watch on the website directly.
```